server/compactor/src/main/java/org/apache/accumulo/compactor/Compactor.java

```diff
@@ -426,8 +426,12 @@ protected void updateCompactionCompleted(TExternalCompactionJob job, TCompaction
    * @throws RetriesExceededException thrown when retries have been exceeded
    */
   protected TExternalCompactionJob getNextJob(Supplier<UUID> uuid) throws RetriesExceededException {
+    final long startingWaitTime =
+        getConfiguration().getTimeInMillis(Property.COMPACTOR_JOB_WAIT_TIME);
+    final long maxWaitTime =
+        getConfiguration().getTimeInMillis(Property.COMPACTOR_MAX_JOB_WAIT_TIME);
```

Comment:
Not sure if this check is needed, maybe redundant.  Maybe RetryableThriftCall or Retry is already doing this check. 

```suggestion
        getConfiguration().getTimeInMillis(Property.COMPACTOR_MAX_JOB_WAIT_TIME);
     Preconditions.checkArgument(   startingWaitTime <= maxWaitTime,  "TODO Message including props names");
```

Reply:
It is being done in `Retry.maxWait`.